build-system/test-configs/forbidden-terms.js

```diff
@@ -0,0 +1,1417 @@
+  });
+  // Single line comments either on its own line or following a space,
+  // semi-colon, or closing brace
+  return contents.replace(/( |}|;|^) *\/\/.*/g, '$1');
+}
+
+/**
+ * Collects any forbidden terms (regex patterns) in the contents of a file,
+ * and provides any possible fix information.
+ *
+ * @param {string} srcFile
+ * @param {string} contents
+ * @param {!Object<string, string|!ForbiddenTermDef>} terms
+ * @return {Array<!ForbiddenTermMatchDef>}
+ */
+function matchForbiddenTerms(srcFile, contents, terms) {
+  const contentsWithoutComments = stripComments(contents);
+  return Object.entries(terms)
+    .map(([term, messageOrDef]) => {
+      const {message, allowlist = null, checkInTestFolder = false} =
+        typeof messageOrDef === 'string'
+          ? {message: messageOrDef}
+          : messageOrDef;
+      // NOTE: we could do a glob test instead of exact check in the future
+      // if needed but that might be too permissive.
+      if (
+        isInBuildSystemFixtureFolder(srcFile) ||
+        (Array.isArray(allowlist) &&
+          (allowlist.indexOf(srcFile) != -1 ||
+            (isInTestFolder(srcFile) && !checkInTestFolder)))
+      ) {
+        return [];
+      }
+
+      const matches = /** @type {Array<!ForbiddenTermMatchDef>} */ ([]);
+      // we can't optimize building the `RegExp` objects early unless we build
+      // another mapping of term -> regexp object to be able to get back to the
+      // original term to get the possible fix value. This is ok as the
+      // presubmit doesn't have to be blazing fast and this is most likely
+      // negligible.
+      const regex = new RegExp(term, 'gm');
+      let index = 0;
+      let line = 1;
+      let column = 0;
+      const start = {line: -1, column: -1};
+
+      let result;
+      while ((result = regex.exec(contentsWithoutComments))) {
+        const [match] = result;
+
+        for (index; index < result.index + match.length; index++) {
+          if (index === result.index) {
+            start.line = line;
+            start.column = column;
+          }
+          if (contentsWithoutComments[index] === '\n') {
+            line++;
+            column = 0;
+          } else {
+            column++;
+          }
+        }
+
+        matches.push({
+          match,
+          term,
+          message,
+          loc: {start, end: {line, column}},
+        });
+      }
+
+      return matches;
+    })
+    .reduce((a, b) => a.concat(b));
+}
+
+/**
+ * @param {string} srcFile
+ * @param {string} contents
+ * @return {Array<!ForbiddenTermMatchDef>}
+ */
+function getForbiddenTerms(srcFile, contents) {
+  const basename = path.basename(srcFile);
+
+  const terms = matchForbiddenTerms(srcFile, contents, forbiddenTerms);
+
+  const isTestFile =
+    /^test-/.test(basename) ||
+    /^_init_tests/.test(basename) ||
+    /_test\.js$/.test(basename) ||
+    /testing\//.test(srcFile) ||
+    /storybook\/[^/]+\.js$/.test(srcFile);
```

Comment:
For long-term maintainability, I wonder if this list can be extracted from info already captured in `test-configs/config.js`. Definitely not a blocker for this PR.

Reply:
This sounds like eslint config to me 🤷‍♂️  In any case, I think we can handle it on a later PR.